src/vs/workbench/browser/actions/layoutActions.ts

```diff
@@ -678,7 +678,7 @@ export class MoveFocusedViewAction extends Action {
 			.map(viewletId => {
 				return {
 					id: viewletId,
-					label: this.viewDescriptorService.getViewContainerById(viewletId)!.title
+					label: this.viewDescriptorService.getViewContainerModel(this.viewDescriptorService.getViewContainerById(viewletId)!)!.title
```

Comment:
Why do we get the view container model now? Was the previous call returning the ugly title?

Reply:
View Container's can change their title dynamically when views are moved into them. The container model provides the current display title for the view container. The `container.title` which was used before only has the statically declared title, which for "generated views" is meaningless (hence the ugly string).